lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -785,7 +787,8 @@ def quickchip_payment?(payment)
       end
 
       def card_from_response(response)
-        response['card'] || response['active_card'] || response['source'] || {}
+        # StripePI puts the AVS and CVC check significantly deeper into the response object
+        response['card'] || response['active_card'] || response['source'] || response.dig('charges', 'data', 0, 'payment_method_details', 'card', 'checks') || {}
```

Comment:
Is `response.charges.data` always an array?

Reply:
all of the verification I did while testing had `response.charges.data` as an array (of a single object) yes